torch/csrc/jit/script/compiler.cpp

```diff
@@ -1691,10 +1691,29 @@ struct to_ir {
       return emitBuiltinCall(
                  loc, *graph, aten::select, {gatherable, idx}, {}, true);
 
-    } else {
-      JIT_ASSERT(gatherable.value(*graph)->type()->isSubtypeOf(DynamicType::get()));
+    } else if (gatherable.value(*graph)->type()->isSubtypeOf(DynamicType::get())) {
       return emitSelect(
           loc, gatherable.value(*graph), /*dim=*/0, idx.value(*graph));
+    } else if (auto tuple = gatherable.value(*graph)->type()->cast<TupleType>()) {
+      auto idx_value = idx.value(*graph);
+      if (idx_value->node()->kind() != prim::Constant || !idx_value->type()->isSubtypeOf(IntType::get())) {
+        throw ErrorReport(loc)
+          << "Indexing on tuples only supported with integer constants";
+      }
+
+      auto index = (*toIValue(idx_value->node()->output())).toInt();
+      if (index >= tuple->elements().size()) {
```

Comment:
You didn't check for negative indices (they're not supported too)

Reply:
Do you think we should support negative indices ?